macrosynergy/dataquery/api.py

```diff
@@ -83,26 +85,26 @@ def check_connection(self) -> Tuple[bool, dict]:
         """Check connection (heartbeat) to DataQuery.
         """
         endpoint = "/services/heartbeat"
-        js: dict = self.access.get_dq_api_result(
+        js, success, msg = self.access.get_dq_api_result(
             url=self.access.base_url + endpoint,
             params={"data": "NO_REFERENCE_DATA"},
             proxy=self.proxy
         )
 
```

Comment:
Since `check_connection()` should return a boolean, I think it makes sense to return`(results["code"]==200)`. Since the `results` dict is also returned, processes calling `check_connection()` have access to `results["code"]`.

Reply:
Is it a good idea to rewrite 'check_connection()' such that it returns `Tuple[bool, dict]` with `(results["code"]==200)` & `results` when called as `check_connection(debug=True)`?
We could then set `check_connection()` to return only a boolean when called without args. This would mean keeping `debug=False` by default, or using `self.debug` from the Interface/JPMaQSDownload object.

Reply:
The re-write is done, and has the assert statement as of now. However, I think it's problematic as it causes `check_connection()` to break (and thus the rest of the code) and not return False. 
So the only valid state for `check_connection()`'s return data would become `[True, {results}]`